src/urllib3/__init__.py

```diff
@@ -99,6 +99,9 @@ def request(
     redirect: Optional[bool] = True,
     retries: Optional[Union[Retry, bool, int]] = None,
     timeout: Optional[Union[Timeout, float, int]] = 3,
+    chunked: bool = False,
+    body_pos: Optional[Union[int, object]] = None,
```

Comment:
I'm unsure if we need `body_pos` and `assert_same_host` defined on the high-level helper as both are low level concepts. `body_pos` is used to signal initial body location in the case of rewinding on redirects, `assert_same_host` only matters at the `ConnectionPool` level or lower (`PoolManager` actually sets `assert_same_host=False` automatically in `urlopen`).

At this high level `redirects=True|False` is the appropriate mechanism to control cross-host redirects.